src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -547,16 +547,17 @@ class Program
                 );
         }
 
-        [ClrOnlyFact(ClrOnlyReason.MemberOrder)]
+        [Fact]
         public void TestNameofIndexerName()
         {
-            var source = @"
+            var t = typeof(NameofTests);
+            var source = @"using System.Linq;
 class C
 {
     public static void Main(string[] args)
     {
         var t = typeof(C);
-        foreach (var m in t.GetMethods())
+        foreach (var m in t.GetMethods().Where(m => m.DeclaringType == t).OrderBy(m => m.Name))
```

Comment:
>.Where(m => m.DeclaringType == t) [](start = 40, length = 33)

I think this can be achieved by including ```DeclaredOnly.BindingFlags``` into flags passed to ```GetMethods```